src/elfpy/pricing_models/base.py

```diff
@@ -201,15 +208,46 @@ def calc_spot_price_from_reserves(
         float
             The spot price of principal tokens.
         """
+        return float(self._calc_spot_price_from_reserves(market_state=market_state, time_remaining=time_remaining))
+
+    def _calc_spot_price_from_reserves(
+        self,
+        market_state: MarketState,
+        time_remaining: StretchedTime,
+    ) -> Decimal:
+        r"""
+        Calculates the spot price of base in terms of bonds.
+
+        The spot price is defined as:
+
+        .. math::
+            \begin{align}
+            p = (\frac{y + cz}{\mu z})^{-\tau}
```

Comment:
shouldn't the denominator be mu * x?

Reply:
Yeah, now that I look at this, I think the whole calculation is off. The numerator is also wrong (should at least be 2y + cz), but something else that seems off is that in the Allen Niemberg paper, they have $p = \frac{y}{\mu x}^t$, which would mean that our calculation should be: $p = (\frac{c \cdot (y + s)}{\mu z})^t \approx (\frac{c \cdot (2y + cz)}{\mu z})^t$. Let me know if this looks right to you, and I'll fix it. We also need to start using LP tokens in the price calculations rather than $s \approx 2y + cz$ but I'll create an issue for that.

Reply:
ya the reason it is to the power of t is b/c they look at it in terms of swapping base for PT.  I/we prefer to look at it the other way b/c it results in spot prices < 1

Reply:
I'm going to resolve this since I updated the comment. As discussed offline, adding an extra `c` into the numerator makes the `without_fee_or_slippage` diverge further from the `without_fee` calculation, so it's a less accurate approximation. Thus, I'm keeping the spot price calculation the same.